### PR TITLE
Where applicable, update Windows pools used to `azsdk-pool-mms-win-2022-general` and rename `vmImage` to the `windows-20xx` format

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
This PR is a follow-up to:
- #3115

Where applicable, update Windows pools used to `azsdk-pool-mms-win-2022-general` and rename `vmImage` to the `windows-20xx` format.

This discussion explains why we chose given `vmImage` format:

[Mike Harder: 1ES Hosted Pool image name changes](https://teams.microsoft.com/l/message/19:59dbfadafb5e41c4890e2cd3d74cc7ba@thread.skype/1676491855184?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1676491855184&teamName=Azure%20SDK&channelName=Engineering%20System%20%F0%9F%9B%A0%EF%B8%8F&createdTime=1676491855184)
posted in Azure SDK / Engineering System 🛠️ at Wednesday, February 15, 2023 12:10 PM

For further context, please see:
- https://github.com/Azure/azure-sdk-tools/issues/3407

### CODE REVIEW

**Comment by konrad-jamrozik** [comment]
@mikeharder what to do about lines like [this one](https://github.com/Azure/autorest.csharp/blob/7bd024de629af5a1d35352a5f6e5d576e615536e/eng/pipelines/cadl-ci.yml#L101)?

> pool: server

Convert it to Ubuntu or Windows?

File: src/assets/Azure.Core.Shared/DiagnosticScope.cs [hunk 1]
```diff
@@ -207,25 +207,6 @@ public ActivityAdapter(object? activitySource, DiagnosticSource diagnosticSource
                 _activityName = activityName;
                 _kind = kind;
                 _diagnosticSourceArgs = diagnosticSourceArgs;
-
```
Comment (konrad-jamrozik) [inline src/assets/Azure.Core.Shared/DiagnosticScope.cs:210]: FYI this change comes from build failure:

> D:\a\_work\1\s\autorest.csharp\eng\SharedCodeCheck.ps1 : Shared source files are updated. Please run eng/DownloadSharedSource.ps1

So I did run `eng/DownloadSharedSource.ps1` locally and pushed the resulting change.

File: src/assets/Azure.Core.Shared/DiagnosticScope.cs [hunk 2]
```diff
@@ -310,6 +291,25 @@ public void AddLink(string traceparent, string tracestate, IDictionary<string, s
...
+                        case ActivityKind.Client:
+                            AddTag("kind", "client");
+                            break;
+                        case ActivityKind.Producer:
+                            AddTag("kind", "producer");
+                            break;
+                        case ActivityKind.Consumer:
+                            AddTag("kind", "consumer");
+                            break;
+                    }
```
Comment (konrad-jamrozik) [inline src/assets/Azure.Core.Shared/DiagnosticScope.cs:311]: FYI this change comes from build failure:

> D:\a\_work\1\s\autorest.csharp\eng\SharedCodeCheck.ps1 : Shared source files are updated. Please run eng/DownloadSharedSource.ps1

So I did run `eng/DownloadSharedSource.ps1` locally and pushed the resulting change.

**Review by mikeharder — APPROVED** [review]

**Comment by mikeharder** [comment]
> @mikeharder what to do about lines like [this one](https://github.com/Azure/autorest.csharp/blob/7bd024de629af5a1d35352a5f6e5d576e615536e/eng/pipelines/cadl-ci.yml#L101)?
> 
> > pool: server
> 
> Convert it to Ubuntu or Windows? I mean, eventually. For now I am tempted to just leave it be.

Any `server` jobs should be left alone.  They are the best kind of job that doesn't even require an agent:

https://learn.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml#server-jobs

**Review by m-nash — APPROVED** [review]

**Pull request merged**